internal/twirptest/service_test.go

```diff
@@ -574,6 +575,48 @@ func TestErroringHooks(t *testing.T) {
 	})
 }
 
+func TestInterceptor(t *testing.T) {
+	interceptor := func(next twirp.Method) twirp.Method {
+		return func(ctx context.Context, request interface{}) (interface{}, error) {
+			size, ok := request.(*Size)
+			if !ok {
+				return nil, fmt.Errorf("could not cast %T to a *Size", request)
+			}
+			size.Inches = size.Inches + 1
+			response, err := next(ctx, request)
+			hat, ok := response.(*Hat)
+			if ok && hat != nil {
+				hat.Color = hat.Color + "x"
+				return hat, err
+			}
+			return nil, err
+		}
+	}
+	h := PickyHatmaker(3)
+
+	s := httptest.NewServer(
+		NewHaberdasherServer(
+			h,
+			twirp.WithServerInterceptors(
+				interceptor,
+				interceptor,
+			),
+		),
+	)
+	defer s.Close()
+	client := NewHaberdasherProtobufClient(s.URL, http.DefaultClient)
+	hat, clientErr := client.MakeHat(context.Background(), &Size{Inches: 1})
+	if clientErr != nil {
+		t.Fatalf("client err=%q", clientErr)
+	}
+	if hat.Size != 3 {
+		t.Errorf("hat size expected=3 actual=%v", hat.Size)
+	}
+	if hat.Color != "bluexx" {
+		t.Errorf("hat color expected=bluexx actual=%v", hat.Color)
+	}
+}
```

Comment:
Make sure to add test cases with no errors (using `&Size{Inches: 3}` on the `PickyHatmaker(3)` should return success with no errors. And also verify that calling with a different size (e.g. `&Size{Inches: 1}`) returns the proper error that can be casted to a `twirp.Error` type and the `twerr.Code()` is `twirp.InvalidArgument` (as returned by the PickyHatmaker fake implementation.

Reply:
Done